docs/docs/config-reference.md

```diff
@@ -3,120 +3,414 @@ id: config-reference
+
+```toml
+merge.automerge_label = "🚀 merge it!"
+```
+
+### `merge.require_automerge_label`
+
+- **type:** `boolean`
+- **default:** `true`
+
+Require that the automerge label (`merge.automerge_label`) be set for Kodiak to merge a PR.
+
+When disabled, Kodiak will immediately attempt to merge any PR that passes all GitHub branch protection requirements.
+
+### `merge.blacklist_title_regex`
+
+- **type:** `string`
+- **default:** `"^WIP:.*"`
+- **options:** Regex pattern or `""`
+
+If a PR's title matches this regex, Kodiak will not merge the PR. This is useful
+to prevent merging work-in-progress PRs.
+
+Setting `merge.blacklist_title_regex = ""` disables this option.
+
+#### example
+
+```
+merge.blacklist_title_regex = ".*DONT\s*MERGE.*"
+```
+
+### `merge.blacklist_labels`
+
+- **type:** `string[]`
+- **default:** `[]`
+- **options:** List of label names
+
+Kodiak will not merge a PR with any of these labels.
+
+#### example
+
+```
+merge.blacklist_labels = ["wip"]
+```
+
+### `merge.method`
+
+- **type:** `string`
+- **default:** `"merge"`
+- **options:** `"merge"`, `"squash"`, `"rebase"`
+
+Choose merge method for Kodiak to use.
+
+Kodiak will report a configuration error if the selected merge method is disabled for a repository.
+
+If you're using the "Require signed commits" GitHub Branch Protection setting to require commit signatures, _`"merge"` is the only compatible option_. Any other option will cause Kodiak to raise a configuration error.
+
+### `merge.delete_branch_on_merge`
+
+- **type:** `boolean`
+- **default:** `false`
+
+Once a PR is merged, delete the branch.
+
+This option behaves like the GitHub repository setting "Automatically delete head branches", which automatically deletes head branches after pull requests are merged.
+
+### `merge.block_on_reviews_requested`
+
+- **type:** `boolean`
+- **default:** `false`
+
+> **DEPRECATED**
+>
+> Due to limitations with the GitHub API this feature is fundamentally broken and cannot be fixed. Prefer the GitHub branch protection "required reviewers" setting instead.
+>
+> When a user leaves a comment on a PR, GitHub counts that as satisfying a review request, so the PR will be allowed to merge, even though a reviewer was likely just starting a review.
+>
+> See this issue comment for more information: [chdsbd/kodiak#153 (comment)](https://github.com/chdsbd/kodiak/issues/153#issuecomment-523057332)
+
+If you request review from a user, don't merge until that user provides a review, even if the PR is passing all status checks.
+
+### `merge.notify_on_conflict`
+
+- **type:** `boolean`
+- **default:** `true`
+
+If there is a merge conflict, make a comment on the PR and remove the
+automerge label.
+
+This option only applies when `merge.require_automerge_label` is enabled.
+
+### `merge.optimistic_updates`
+
+- **type:** `boolean`
+- **default:** `true`
+
+Don't wait for in-progress status checks on a PR to finish before updating the branch.
```

Comment:
maybe elaborate on why someone would want this

something like `This setting can speed up merges but increases chance of running CI unnecessarily`